dts/bindings/ethernet/st,stm32-ethernet-parent.yaml

```
@@ -0,0 +1,16 @@
+# Copyright The Zephyr Project Contributors
+# SPDX-License-Identifier: Apache-2.0
+
+description: ST STM32 Ethernet parent node
+
+compatible: "st,stm32-ethernet-parent"
```

What do you think about renaming the existing `st,stm32<XXX>-ethernet` to `st,stm32<XXX>-ethernet-mac` then naming this binding/node:
```suggestion
description: STM32 Ethernet controller

compatible: "st,stm32-ethernet"
```

(`st,stm32-mdio` could also be renamed `st,stm32-ethernet-mdio` but that's cosmetic)

---

I'm unsure, I'm more on the side of not renaming too much.

For mdio, I didn't find a separate mdio controller in the datasheets and hal, what I instead found was a MDIO slave interface, so it can act as a peripheral on the mdio bus. Currently this functionality is not supported by the zephyr mdio api. And for DT compatible, that should then reflect, that it is not a controller by a describing suffex.

---

You're correct again, I misunderstood RM0399 §57 as MDIO master... ~~in hindsight, everything makes much more sense when you consider the IP is a MDIO slave 😅~~

My main dislike point is the `ethernet-parent` name... Would renaming this binding to `st,stm32-ethernet-controller` - without changing the other bindings - be acceptable for you?
(We could add a note in the `st,stm32<XXX>-ethernet` bindings' description indicating they are for the MAC)